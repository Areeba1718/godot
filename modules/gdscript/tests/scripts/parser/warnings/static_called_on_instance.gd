class Player:
	var x = 3

static func box(x : int, y : int, z : int, x_max : int, y_max : int) -> int:
	return (z * x_max * y_max) + (y * x_max) + x
	
static func boxv(v : Vector3i, x_max : int, y_max : int) -> int:
	return box(v.x, v.y, v.z, x_max, y_max)

func test():
	# These should not emit a warning.
	var _player = Player.new()
	print(String.num_uint64(8589934592)) # 2 ^ 33

	# This should emit a warning.
	var some_string = String()
	print(some_string.num_uint64(8589934592)) # 2 ^ 33

	# These should not emit a warning.
	Globals.boxv(Vector3i(0, 0, 0), 10, 10)
