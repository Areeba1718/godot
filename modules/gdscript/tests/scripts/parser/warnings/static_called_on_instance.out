GDTEST_OK
>> WARNING
>> Line: 17
>> STATIC_CALLED_ON_INSTANCE
>> The function "num_uint64()" is a static function but was called from an instance. Instead, it should be directly called from the type: "String.num_uint64()".
>> WARNING
>> Line: 20
>> STATIC_CALLED_ON_INSTANCE
>> The function "boxv()" is a static function but was called from an instance. Instead, it should be directly called from the type: "RefCounted.boxv()".
8589934592
8589934592
